### PR TITLE
travis: Disable source tarballs on most builders

### PR DESCRIPTION
**Pull request by alexcrichton**
Currently we create a source tarball on almost all of the `DEPLOY=1` builders
but this has the adverse side effect of all source tarballs overriding
themselves in the S3 bucket. Normally this is ok but unfortunately a source
tarball created on Windows is not buildable on Unix.

On Windows the vendored sources contain paths with `\` characters in them which
when interpreted on Unix end up in "file not found" errors.

Instead of this overwriting behavior, whitelist just one linux builder for
producing tarballs and avoid producing tarballs on all other hosts.

### CODE REVIEW

File: src/ci/docker/dist-x86-linux/Dockerfile [hunk 1]
```diff
@@ -82,5 +82,10 @@ RUN curl -L https://api.pub.build.mozilla.org/tooltool/sha512/$SCCACHE_DIGEST |
 ENV HOSTS=i686-unknown-linux-gnu
 ENV HOSTS=$HOSTS,x86_64-unknown-linux-gnu
 
-ENV RUST_CONFIGURE_ARGS --host=$HOSTS --enable-extended --enable-sanitizers
+ENV RUST_CONFIGURE_ARGS \
+      --host=$HOSTS \
+      --enable-extended \
+      --enable-sanitizers \
+      --disable-dist-src
```
Comment (eddyb) [inline src/ci/docker/dist-x86-linux/Dockerfile]: I'm confused, why is this line here?

Comment (alexcrichton) [inline src/ci/docker/dist-x86-linux/Dockerfile]: oops I'm not thinking correctly

**Comment by frewsxcv** [comment]
It looks like this PR might have caused this error:

https://travis-ci.org/rust-lang/rust/jobs/202083203

**Comment by alexcrichton** [comment]
@bors: r-

indeed quite likely!
